spackbot/handlers/pipelines.py

```diff
@@ -52,20 +52,43 @@ async def run_pipeline(event, gh):
     branch = pr["head"]["ref"]
     branch = f"github/pr{number}_{branch}"
     branch = urllib.parse.quote_plus(branch)
-
-    url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
-    headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
-
-    logger.info(f"{sender} triggering pipeline, url = {url}")
-
-    # Don't provide GitHub credentials to GitLab!
-    async with aiohttp.ClientSession() as session:
-        async with session.post(url, headers=headers) as response:
-            result = await response.json()
-
+    if not retry_pipeline:
+        trigger_type = "started"
+        url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
+        headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
+
+        # Don't provide GitHub credentials to GitLab!
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers) as response:
+                result = await response.json()
+    else:
+        # Get a list of failed pipelines
+        trigger_type = "retried"
+        query_string = f"?status=failed&order_by=updated_at&ref={branch}"
+        url = f"{helpers.gitlab_spack_project_url}/pipelines{query_string}"
+        headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
+
+        # Don't provide GitHub credentials to GitLab!
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers) as response:
+                pipelines = await response.json()
+        logger.info(f"{sender} triggering pipeline, url = {url}")
+
+        if pipelines and "error" not in pipelines:
+            # Index 0 has the latest pipeline which has been marked as failed
```

Comment:
Based on the default value of the `sort` query param described [here](https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines), I think I agree index 0 has the latest `failed` pipeline.